contracts/finance/VestingWallet.sol

```diff
@@ -133,14 +133,36 @@ contract VestingWallet is Context, Ownable {
      * @dev Calculates the amount of ether that has already vested. Default implementation is a linear vesting curve.
      */
     function vestedAmount(uint64 timestamp) public view virtual returns (uint256) {
-        return _vestingSchedule(address(this).balance + released(), timestamp);
+        uint256 balance = address(this).balance;
+        uint256 releasedAmount = released();
+
+        // Check for overflow and cap at type(uint256).max
+        uint256 totalAllocation = balance;
+        if (totalAllocation <= type(uint256).max - releasedAmount) {
+            totalAllocation += releasedAmount;
+        } else {
+            totalAllocation = type(uint256).max;
+        }
+
+        return _vestingSchedule(totalAllocation, timestamp);
```

Comment:
If we want to do that change (I'm not sure we do), then we should implement it using `saturatingAdd`.

We have tools to do that computation in a readable and gas efficient way. No need to add branches and (unchecked) logic like that.
```suggestion
        return _vestingSchedule(address(this).balance.saturatingAdd(released()), timestamp);
```